challenges/lib/src/main/java/challenges/tree/BinarySearchTree.java

```diff
@@ -0,0 +1,39 @@
+package challenges.tree;
+
+import java.util.ArrayList;
+
+public class BinarySearchTree extends BinaryTree{
+
+
+    public void add (int input){
+        if (this.root == null) this.root = new Node(input);
+        else{
+            boolean done = false;
+            Node currentNode = this.root;
+            while (done == false){
+                if ( input < currentNode.val){
+                    if (currentNode.left == null){
+                        currentNode.left = new Node(input);
+                        done = true;
+                    } else currentNode = currentNode.left;
+                }else if (input == currentNode.val){

```

Comment:
So line 19 says if the given input is == current value, don't add it to the bst?